src/Validator/Constraints/PhoneNumber.php

```diff
@@ -43,54 +43,82 @@ class PhoneNumber extends Constraint
     public $type = self::ANY;
     public $defaultRegion = PhoneNumberUtil::UNKNOWN_REGION;
 
-    public function getType()
+    public function getType(): ?string
     {
-        switch ($this->type) {
-            case self::FIXED_LINE:
-            case self::MOBILE:
-            case self::PAGER:
-            case self::PERSONAL_NUMBER:
-            case self::PREMIUM_RATE:
-            case self::SHARED_COST:
-            case self::TOLL_FREE:
-            case self::UAN:
-            case self::VOIP:
-            case self::VOICEMAIL:
-                return $this->type;
+        @trigger_error(__METHOD__.' is deprecated and will be removed in 4.0. Use `getTypes` instead.', E_USER_DEPRECATED);
+
+        $types = $this->getTypes();
+        if (0 === count($types)) {
+            return null;
+        }
+
+        return reset($types);
+    }
+
+    public function getTypes(): array
+    {
+        if (is_array($this->type)) {
```

Comment:
keeping `type` (without `s`) to store a array is weird. Shouldn't we deprecate it too?

Reply:
I'd like to but it's a public property, renaming it right now will cause a BC Break. :/

Reply:
that's why we should provide a migration path to be able to remove it in 4.0

ie. providing both `type` and `types`, merging data from both properties in `getTypes`. and triggering a deprecation if `type` is not null.

Reply:
In fact, I think it makes sense to keep the existing parameter name.
Current usage of this class is this one:
```
 * @AssertPhoneNumber(type="mobile")
```
If we decide to rename the property to types, it will force everyone to update their usage of this assertion. The current code introduce a BC break but a "small" one: you're not supposed to use the `getType` method in your code.
Updating the variable is a "huge" BC break: all end users using this validator will have to update their code.
Furthermore, when using `$types` instead of `$type`. 
* either we allow end users to specifiy only 1 type without using an argument (`types="mobile`). In this case, it's not coherent with the `types` name but we're closer of the actual usage.
* either everyone will have to use an array (`types={"mobile"}`). It's obviously more coherent but it forces you to use an array when you don't need it (an today, nobody need it as it's not supported).

IMO, we should keep the property`$type` which can either contain a single type (no change at all for end users) or an array of types (new feature introduced by this PR).
WDYT?